File: App.client/projects/proctor-app/scripts/models/audit-log.ts
import { User } from './user';

export class AuditLog {
  id: number;
  userId?: number | null;
  action: string;
  entityType: string;
  entityId: number;
  oldValues?: string | null;
  newValues?: string | null;
  ipAddress?: string | null;
  userAgent?: string | null;
  createdAt: string;
  createdOn: string;
  modifiedOn: string;
  user?: User | null;

  constructor() {
    this.id = 0;
    this.userId = null;
    this.action = this.entityType = '';
    this.entityId = 0;
    this.oldValues = this.newValues = this.ipAddress = this.userAgent = null;
    this.createdAt = new Date().toISOString();
    this.createdOn = this.modifiedOn = '';
  }
}

File: App.Models/AuditLog.cs
using App.Core.Models;

namespace App.Models
{
    public class AuditLog : IAuditableEntity
    {
        public int Id { get; set; }
        public int? UserId { get; set; }
        public string Action { get; set; } = string.Empty;
        public string EntityType { get; set; } = string.Empty;
        public int EntityId { get; set; }
        public string? OldValues { get; set; }
        public string? NewValues { get; set; }
        public string? IpAddress { get; set; }
        public string? UserAgent { get; set; }
        public DateTime CreatedAt { get; set; }
        public DateTime CreatedOn { get; set; }
        public DateTime ModifiedOn { get; set; }

        // Navigation properties
        public virtual User? User { get; set; }
    }
}


